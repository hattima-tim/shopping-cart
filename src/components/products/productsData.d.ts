export interface Product {
  path?:string;
  breadCrumbs: Array<{ name: string; path: string }>;
  pathName?: string;
  fullPath: string;
  imagesForProductPage: string[];
  imgForProductCard?: string;
  name: string;
  price: string;
  details: string[];
  fabric: string[];
  color: string[];
  size: string[];
  description: string[];
  additionalInfo: {
    weight: string;
    fabric?: string;
    size: string;
  };
}