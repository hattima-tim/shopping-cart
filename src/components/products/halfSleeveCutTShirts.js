import { Link } from "react-router-dom";
import '../../styles/productsPage.css';
const HalfSleeveCutTShirts = () => {
    return (
        <div className='breadCrumbs'>
           <Link to='/'>Home</Link> / Half Sleeve Cut T-Shirts
        </div>
    )
}

export default HalfSleeveCutTShirts;