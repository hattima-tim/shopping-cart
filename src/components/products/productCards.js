import { Link } from "react-router-dom";

function ProductCards({ products }) {
  return (
    <div className="productCards">
      {products.map((product, index) => (
        <div className="productCard" key={index}>
          <div className="productCardImg">
            <Link to={`/product/${product.pathName}`}>
              <img src={product.img} alt={product.name} />
            </Link>
          </div>
          <div className="productCard__info">
            <h3>{product.name}</h3>
            <p>{product.price}</p>
          </div>
        </div>
      ))}
    </div>
  );
}

export default ProductCards;
