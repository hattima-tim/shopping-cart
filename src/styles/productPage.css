.main{
    display: flex;
    flex-wrap: wrap;
}

.productImage{
    object-fit: cover;
}

.mainInfo{
    flex: 1;
    padding: 0 1rem;
}

.productOptionBtn{
    margin: 0.16rem;
    width: fit-content;
    background: transparent;
    border: 1px solid #A8A8A8;
    border-radius: 3px ;
}

.productOptionBtn:hover{
    border: 2px solid black;
}

