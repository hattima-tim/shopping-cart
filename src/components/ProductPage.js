import { useState } from "react";
import { Outlet, useParams, Link, useOutletContext } from "react-router-dom";
import uniqid from "uniqid";
import ImageMagnifier from "./imageMagnifier";
import "../styles/productPage.css";

export function Product() {
  return (
    <>
      <Outlet context={useOutletContext()} />
    </>
  );
}

function ProductPage({ getProductData }) {
  const params = useParams();
  const productData = getProductData(params.name);

  const [itemNumber, setItemNumber] = useState(1);
  const itemNumberIncrement = () => {
    setItemNumber(itemNumber + 1);
  };
  const itemNumberDecrement = () => {
    if (itemNumber > 1) {
      setItemNumber(itemNumber - 1);
    }
  };
  const handleChange = (e) => {
    if (e.target.value > 0) {
      const value = parseInt(e.target.value);
      setItemNumber(value);
    }
  };

  const [currentlyShowingAdditionalInfo, setCurrentlyShowingAdditionalInfo] =
    useState("description");

  const styleTabAsActive = () => {
    const descriptionTab = document.querySelector(".description");
    descriptionTab.classList.toggle("active-tab");

    const additionalInfoTab = document.querySelector(".additional-info");
    additionalInfoTab.classList.toggle("active-tab");
  };

  const showDescription = () => {
    setCurrentlyShowingAdditionalInfo("description");
    styleTabAsActive();
  };
  const showAdditionalInfo = () => {
    setCurrentlyShowingAdditionalInfo("additionalInfo");
    styleTabAsActive();
  };

  const [productsInCart, setProductsInCart] = useOutletContext();

  const [fabric, setFabric] = useState("");

  const handleFabricBtnClick = (e) => {
    const currentBtn = e.target;
    const isButtonActive = currentBtn.classList.contains("active");

    const fabricBtns = document.querySelectorAll(".fabric-btn");
    fabricBtns.forEach((btn) => {
      if (btn !== currentBtn) {
        btn.classList.remove("active");
      }
    });

    if (isButtonActive) {
      // means that this button is pressed twice to deselect it
      currentBtn.classList.remove("active");
      setFabric("");
    } else {
      currentBtn.classList.add("active");
      setFabric(currentBtn.textContent);
    }
  };

  const [size, setSize] = useState("");

  const handleSizeBtnClick = (e) => {
    const currentBtn = e.target;
    const isButtonActive = currentBtn.classList.contains("active");

    const sizeBtns = document.querySelectorAll(".size-btn");
    sizeBtns.forEach((btn) => {
      if (btn !== currentBtn) {
        btn.classList.remove("active");
      }
    });

    if (isButtonActive) {
      // means that this button is pressed twice to deselect it
      currentBtn.classList.remove("active");
      setSize("");
    } else {
      currentBtn.classList.add("active");
      setSize(currentBtn.textContent);
    }
  };

  let product = {
    id: uniqid(),
    name: productData.name,
    img: productData.img,
    price: productData.price,
    quantity: itemNumber,
    subTotal: itemNumber * (productData.price.split("৳")[1] * 1),
    fabric: fabric,
    size: size,
    path: params.name,
  };

  const handleAddToCart = () => {
    if (size === "" || fabric === "") {
      alert("Please select both fabric and size");
      return;
    }
    const newProduct = { ...product };
    localStorage.setItem(
      "productsInCart",
      JSON.stringify([...productsInCart, newProduct])
    );
    setProductsInCart([...productsInCart, newProduct]);
  };

  return (
    <div className="lg:mx-16">
      <div className="main mx-4 mb-12 mt-4 lg:flex">
        <ImageMagnifier
          src={productData.imgForProductPage}
          alt={productData.name}
        />

        <div className="mainInfo lg:ml-8">
          <div className="breadcrumbs text-sm lg:text-base">
            {productData.breadCrumbs.map((breadCrumb, index) => {
              if (index === productData.breadCrumbs.length - 1) {
                return (
                  <Link key={uniqid()} to={breadCrumb.path}>
                    {breadCrumb.name}
                  </Link>
                );
              } else {
                return (
                  <Link key={uniqid()} to={breadCrumb.path}>
                    {breadCrumb.name + ' / '}
                  </Link>
                );
              }
            })}
          </div>

          <h1 className="mt-2 mb-6 text-xl font-bold text-gray-600 lg:text-3xl">
            {productData.name}
          </h1>
          <h2 className="mb-4 text-2xl font-bold text-slate-800">
            {productData.price}
          </h2>
          <p className="mb-6">
            <strong>Detailed Specification</strong>
          </p>
          <ul className="details list-disc">
            {productData.details.map((detail, index) => {
              return (
                <li key={index} className="mx-4 my-3">
                  {detail}
                </li>
              );
            })}
          </ul>

          {productData.fabric.toString() !== "" && ( // for some products fabric data is not present
            <div className="fabricButtons">
              <span className="text-sm font-bold text-slate-800">FABRIC</span>
              <br />
              {productData.fabric.map((fabric, index) => {
                return (
                  <button
                    key={index}
                    className="fabric-btn mb-1 mr-2 w-fit rounded-sm border-2 border-black bg-white p-1 text-base text-black"
                    onClick={handleFabricBtnClick}
                  >
                    {fabric}
                  </button>
                );
              })}
            </div>
          )}

          <div className="sizeButtons">
            <span className="text-sm font-bold text-slate-800">SIZE</span>
            <br />
            {productData.size.map((size, index) => {
              return (
                <button
                  key={index}
                  className="size-btn mb-8 mr-2 w-fit rounded-sm border-2 border-black bg-white p-1 text-base text-black"
                  onClick={handleSizeBtnClick}
                >
                  {size}
                </button>
              );
            })}
          </div>

          <div className="addToCart flex gap-4">
            <div className="quantity">
              <button
                onClick={itemNumberDecrement}
                className="h-10 w-8 bg-gray-200 hover:bg-slate-300"
              >
                -
              </button>
              <input
                type="number"
                onChange={handleChange}
                value={itemNumber}
                className="h-10 w-8 border border-y-gray-200 text-center"
              />
              <button
                onClick={itemNumberIncrement}
                className="h-10 w-8 bg-gray-200 hover:bg-slate-300"
              >
                +
              </button>
            </div>

            <button
              onClick={handleAddToCart}
              className="w-fit bg-black px-6 py-2 font-bold text-white "
            >
              Add to Cart
            </button>
          </div>
        </div>
      </div>
      <div className="additional_info m-4">
        <div className="additional_info_tabs md:border-top-gray-400 md:flex md:gap-2 md:border-t">
          <button
            onClick={showDescription}
            className="description active-tab peer border-t py-2 text-start font-bold hover:border-t-2 hover:border-t-black peer-hover:border-t peer-hover:border-t-gray-400"
          >
            Description
          </button>
          <br />
          <button
            onClick={showAdditionalInfo}
            className="additional-info peer border-t py-2 text-start font-bold hover:border-t-2 hover:border-t-black "
          >
            Additional Information
          </button>
        </div>

        {currentlyShowingAdditionalInfo === "description" && (
          <div className="additional_info_content my-4">
            <p>
              <strong>Size Measurement (in inch/centimeter):</strong>
            </p>

            <ul>
              {productData.description.map((description, index) => {
                return (
                  <li key={index} className="my-4">
                    {description}
                  </li>
                );
              })}
            </ul>

            <p className="my-4">
              ডিভাইসের ভিন্নতা অনুযায়ী কালার ডেভিয়েশন এবং ফটোশপ এডিটিং কালার
              ডেভিয়েশনের কারনে বাস্তব ছবি ও সফটকপির মাঝে অনেক সময় পার্থক্য থেকে
              যায়
            </p>
          </div>
        )}
        {currentlyShowingAdditionalInfo === "additionalInfo" && (
          <table className="additional_info_table my-4">
            <tbody>
              <tr>
                <th>Weight</th>
                <td>{productData.additionalInfo.weight}</td>
              </tr>
              <tr>
                <th>Fabric</th>
                <td>{productData.additionalInfo.fabric}</td>
              </tr>
              <tr>
                <th>Size</th>
                <td>{productData.additionalInfo.size}</td>
              </tr>
            </tbody>
          </table>
        )}
      </div>
    </div>
  );
}

export default ProductPage;
