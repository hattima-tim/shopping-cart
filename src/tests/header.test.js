import { render, screen } from "@testing-library/react";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import Header from "../components/header";
import { Product } from "../components/products/productPage";
import ProductPage from "../components/products/productPage";

test("logo image is visible", () => {
  render(<Header />);
  const logo = screen.getByRole("img", { name: "one ummah logo" });
  expect(logo).toBeInTheDocument();
});

const setup = () => {
  const {container} = render(
    <MemoryRouter
      initialEntries={["/product/half-sleeve-cut-and-sew-solid-pattern-15"]}
    >
      <Routes>
        <Route path="/" element={<Header />}>
          <Route path="product" element={<Product />}>
            <Route path=":name" element={<ProductPage />} />
          </Route>
        </Route>
      </Routes>
    </MemoryRouter>
  );

  const fabricBtn = screen.getByRole("button", { name: "COMBED COTTON" });
  userEvent.click(fabricBtn);

  const sizeBtn = screen.getByRole("button", { name: "M" });
  userEvent.click(sizeBtn);

  const quantityBtn = screen.getByRole("button", { name: "+" });
  userEvent.click(quantityBtn);
  userEvent.click(quantityBtn);

  const addToCartBtn = screen.getByRole("button", { name: "Add to Cart" });
  userEvent.click(addToCartBtn);

  return container;
};

test('header and product page', () => {
  const container = setup();
  expect(container).toMatchSnapshot(); // the snapshot contains both
  // Header and ProductPage
})

test("totalItems count is correct", () => {
  setup();
  const totalItem = screen.getByTestId("cart-item-count");
  expect(totalItem).toHaveTextContent("3");
});

describe("products in cart", () => {
  test("remove button removes product from cart", () => {
    setup();

    const productRemoveBtn = screen.getByRole("button", {
      name: "x",
    });
    const productName = screen.getByRole(
      "heading",
      { level: 3 },
      { name: "Half Sleeve Cut and Sew Solid(pattern 15)" }
    );

    userEvent.click(productRemoveBtn);

    expect(productRemoveBtn).not.toBeInTheDocument();
    expect(productName).not.toBeInTheDocument();
  });

});