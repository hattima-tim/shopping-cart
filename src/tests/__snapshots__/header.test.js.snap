// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header header and product page 1`] = `
<div>
  <div>
    <div
      class="header-container"
    >
      <div
        class="header z-10 w-full items-center bg-[#f0f0f0] px-3 md:px-8 lg:justify-between lg:px-8"
      >
        <div
          class="lg:hidden"
        >
          <div
            class="group flex h-8 w-8 cursor-pointer items-center justify-center rounded-full border border-gray-500 p-3 hover:bg-black md:p-0"
          >
            <svg
              class="m-[-1rem] w-3 fill-gray-500 group-hover:fill-white md:m-[.1rem]"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
              />
            </svg>
          </div>
          <div
            class="sidenav z-50 block w-4/6 overflow-auto md:w-7/12"
          >
            <div
              class="closebtn cursor-pointer hover:text-black"
            >
              ×
            </div>
            <ul>
              <li
                class="flex border-b p-3"
              >
                <a
                  class=" text-sm font-semibold"
                  href="/"
                >
                  HOME
                </a>
              </li>
              <li
                class="border-b px-3 py-2 hover:bg-[#e8e8e7]"
              >
                <div
                  class="flex cursor-pointer justify-between "
                >
                  <a
                    class="text-sm font-semibold"
                    href="/"
                  >
                    MEN'S COLLECTION
                  </a>
                  <img
                    alt="down arrow icon"
                    class="down-arrow w-6 cursor-pointer"
                    src="https://res.cloudinary.com/du3oueesv/image/upload/v1662729992/shopping%20cart/polo%20t%20shirts/product%20card%20image/Mask_group_1_mkkawx.png"
                  />
                </div>
                <div
                  class="child-sidenav flex flex-col"
                >
                  <h2
                    class="mx-6 border-b py-2 text-sm font-medium text-black"
                  >
                    HALF SLEEVE T SHIRT
                  </h2>
                  <a
                    class="mx-8 border-b py-2 text-base font-normal"
                    href="/half-sleeve-dawah-tshirts-for-men"
                  >
                    Half Sleeve Dawah T-Shirt
                  </a>
                  <a
                    class="mx-8 border-b py-2 text-base font-normal"
                    href="/half-sleeve-regular-tshirts-for-men"
                  >
                    Half Sleeve Regular T-Shirt
                  </a>
                  <a
                    class="mx-8 border-b py-2 text-base font-normal"
                    href="/half-sleeve-cut-and-sew-solid"
                  >
                    Half Sleeve Cut & Sew T-Shirt
                  </a>
                  <a
                    class="mx-8 border-b py-2 text-base font-normal"
                    href="/polo-t-shirt"
                  >
                    Premium Lascot Polo T-Shirts
                  </a>
                  <h2
                    class="mx-6 border-b py-2 text-sm font-medium text-black"
                  >
                    POLO T-SHIRT
                  </h2>
                  <a
                    class="mx-8 border-b py-2 text-base font-normal"
                    href="/solid-polo-t-shirt"
                  >
                    Solid Polo T-Shirt
                  </a>
                  <a
                    class="mx-8 border-b py-2 text-base font-normal"
                    href="/round-neck-logo-t-shirt"
                  >
                    Round Neck Logo Style
                  </a>
                  <a
                    class="mx-8 border-b py-2 text-base font-normal"
                    href="/cut-and-sew-logo-t-shirt"
                  >
                    Cut & Sew Logo Style
                  </a>
                  <h2
                    class="mx-6 border-b py-2 text-sm font-medium text-black"
                  >
                    JOGGERS COLLECTION
                  </h2>
                  <a
                    class="mx-8 border-b py-2 text-base font-normal"
                    href="/joggers"
                  >
                    Mash Joggers
                  </a>
                </div>
              </li>
              <li
                class="border-b px-3 py-2 hover:bg-[#e8e8e7]"
              >
                <div
                  class="flex cursor-pointer justify-between "
                >
                  <a
                    class="text-sm font-semibold"
                    href="/"
                  >
                    KID'S & BOY'S COLLECTION
                  </a>
                  <img
                    alt="down arrow icon"
                    class="down-arrow w-6 cursor-pointer"
                    src="https://res.cloudinary.com/du3oueesv/image/upload/v1662729992/shopping%20cart/polo%20t%20shirts/product%20card%20image/Mask_group_1_mkkawx.png"
                  />
                </div>
                <div
                  class="child-sidenav flex flex-col"
                >
                  <a
                    class="mx-6 border-b py-2 text-base font-normal"
                    href="/half-sleeve-t-shirt"
                  >
                    Half Sleeve T-Shirt
                  </a>
                </div>
              </li>
              <li
                class="border-b px-3 py-2 hover:bg-[#e8e8e7]"
              >
                <div
                  class="flex cursor-pointer justify-between "
                >
                  <a
                    class="text-sm font-semibold"
                    href="/"
                  >
                    SUNNAH ESSENTIAL
                  </a>
                  <img
                    alt="down arrow icon"
                    class="down-arrow w-6 cursor-pointer"
                    src="https://res.cloudinary.com/du3oueesv/image/upload/v1662729992/shopping%20cart/polo%20t%20shirts/product%20card%20image/Mask_group_1_mkkawx.png"
                  />
                </div>
                <div
                  class="child-sidenav flex flex-col"
                >
                  <a
                    class="mx-6 border-b py-2 text-base font-normal"
                    href="/perfume"
                  >
                    Perfume
                  </a>
                  <a
                    class="mx-6 border-b py-2 text-base font-normal"
                    href="/combo-perfume-box"
                  >
                    Combo Perfume Box
                  </a>
                  <a
                    class="mx-6 border-b py-2 text-base font-normal"
                    href="/prayermat"
                  >
                    PrayerMat
                  </a>
                </div>
              </li>
              <li
                class="flex justify-between border-b p-3"
              >
                <a
                  class="text-sm font-semibold"
                  href="/"
                >
                  FABRIC MASK
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="search-bar ml-[10px]"
          data-testid="searchContainer"
        >
          <div
            class="relative"
          >
            <input
              class="hidden w-80 rounded-full bg-[#e9e9e9] p-2 text-sm lg:block"
              placeholder="Search"
              role="searchbox"
              type="text"
            />
            <img
              alt="searchIcon"
              class="h-4 cursor-pointer w-4 lg:absolute lg:right-2.5 lg:top-[10px]"
              src="https://res.cloudinary.com/du3oueesv/image/upload/v1663124810/shopping%20cart/1890px-Vector_search_icon.svg_odyv9y.png"
            />
          </div>
          <div
            class="search-results absolute z-10 max-h-[100vh] w-full overflow-auto bg-white lg:w-80"
          />
        </div>
        <a
          class="my-3 ml-auto mr-[1rem] lg:mr-64"
          href="/"
        >
          <img
            alt="one ummah logo"
            class="companyLogo w-24"
            src="https://res.cloudinary.com/du3oueesv/image/upload/v1660122044/shopping%20cart/oubd-logo-with-tm-black-1-1400x642_se0t70.png"
          />
        </a>
        <div
          class="shopping-cart-icon ml-auto cursor-pointer px-2 py-2 lg:py-1"
        >
          <span
            class="hidden self-center text-sm text-black lg:inline-block"
            id="subTotal"
          >
            ৳ 
            450
            .00
          </span>
          <span
            class="fa-layers fa-fw"
          >
            <svg
              class="m-0 w-5"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 0C107.5 0 117.4 8.19 119.6 19.51L121.1 32H541.8C562.1 32 578.3 52.25 572.6 72.66L518.6 264.7C514.7 278.5 502.1 288 487.8 288H170.7L179.9 336H488C501.3 336 512 346.7 512 360C512 373.3 501.3 384 488 384H159.1C148.5 384 138.6 375.8 136.4 364.5L76.14 48H24C10.75 48 0 37.25 0 24C0 10.75 10.75 0 24 0H96zM128 464C128 437.5 149.5 416 176 416C202.5 416 224 437.5 224 464C224 490.5 202.5 512 176 512C149.5 512 128 490.5 128 464zM512 464C512 490.5 490.5 512 464 512C437.5 512 416 490.5 416 464C416 437.5 437.5 416 464 416C490.5 416 512 437.5 512 464z"
              />
            </svg>
            <span
              class="fa-layers-counter left-[13px] top-[-12px] font-bold"
              data-testid="cart-item-count"
              style="background: black;"
            >
              1
            </span>
          </span>
        </div>
        <div
          class="cart-drawer"
        >
          <div
            class="closebtn cursor-pointer hover:text-black lg:hidden"
          >
            ×
          </div>
          <h1
            class="my-2 text-center text-xl font-bold lg:hidden"
          >
            Cart
          </h1>
          <div
            class="mx-auto mb-4  w-16 border-2 bg-[#7f7f7f] lg:hidden"
          />
          <div
            class="products_container"
          >
            <div
              class="product_tooltip_item"
            >
              <div
                class="product_tooltip_item_img"
              >
                <img
                  alt="Half Sleeve Cut and Sew Solid(pattern 15)"
                  src="https://res.cloudinary.com/du3oueesv/image/upload/v1661588492/shopping%20cart/half%20sleeve%20cut/mthcssp15-solid-920x920_him69g.jpg"
                />
              </div>
              <div
                class="product_tooltip_item_info"
              >
                <div
                  class="tooltip_header"
                >
                  <h3
                    class="text-[#4e4e4e]"
                  >
                    Half Sleeve Cut and Sew Solid(pattern 15)
                  </h3>
                  <button
                    class="remove_item_btn p-2"
                    data-id="0"
                    data-testid="remove-item-btn"
                  >
                    x
                  </button>
                </div>
                <p>
                  FABRIC: 
                  COMBED COTTON
                </p>
                <p>
                  SIZE: 
                  M
                </p>
                <p>
                  1
                   x
                   
                  <span
                    class="font-bold"
                  >
                    ৳450.00
                  </span>
                </p>
              </div>
            </div>
          </div>
          <p
            class="flex justify-center border-b-2 border-t border-b-[#dcdcdc] border-t-[#dcdcdc] py-3"
          >
            <span>
              Subtotal: ৳
            </span>
            <span
              class="text-black"
            >
              450
            </span>
          </p>
          <a
            class="view_cart"
            href="/cart"
          >
            VIEW CART
          </a>
          <button
            class="checkout_btn"
          >
            CHECKOUT
          </button>
        </div>
      </div>
      <div
        class="align-center hidden w-full items-center justify-center gap-4 bg-black p-5 text-white lg:flex"
        id="top-nav-bar"
      >
        <a
          class="text-sm font-medium text-[#f0f0f0]"
          href="/"
        >
          HOME
        </a>
        <div
          class="group relative flex items-center"
        >
          <a
            class="text-sm font-medium text-[#f0f0f0] hover:text-[#f0f0f0]"
            href="/"
          >
            MEN'S COLLECTION
          </a>
          <svg
            class="my-0 mx-1 w-[.6rem] fill-[#9b9392]"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
            />
          </svg>
          <div
            class="tooltip absolute flex gap-4 group-hover:visible"
          >
            <div
              class="flex flex-col"
            >
              <h2
                class="m-0 border-b py-2 text-sm font-medium text-black"
              >
                HALF SLEEVE T SHIRT
              </h2>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/half-sleeve-dawah-tshirts-for-men"
              >
                Half Sleeve Dawah T-Shirt
              </a>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/half-sleeve-regular-tshirts-for-men"
              >
                Half Sleeve Regular T-Shirt
              </a>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/half-sleeve-cut-and-sew-solid"
              >
                Half Sleeve Cut & Sew T-Shirt
              </a>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/polo-t-shirt"
              >
                Premium Lascot Polo T-Shirts
              </a>
            </div>
            <div
              class="flex flex-col"
            >
              <h2
                class="m-0 border-b py-2 text-sm font-medium text-black"
              >
                Polo T-Shirt
              </h2>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/solid-polo-t-shirt"
              >
                Solid Polo T-Shirt
              </a>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/round-neck-logo-t-shirt"
              >
                Round Neck Logo Style
              </a>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/cut-and-sew-logo-t-shirt"
              >
                Cut & Sew Logo Style
              </a>
            </div>
            <div
              class="flex flex-col"
            >
              <h2
                class="m-0 border-b py-2 text-sm font-medium text-black"
              >
                Joggers Collection
              </h2>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/joggers"
              >
                Mash Joggers
              </a>
            </div>
          </div>
        </div>
        <div
          class="group relative flex items-center"
        >
          <a
            class="text-sm font-medium text-[#f0f0f0] hover:text-[#f0f0f0]"
            href="/"
          >
            KID'S & BOY'S COLLECTION
          </a>
          <svg
            class="my-0 mx-1 w-[.6rem] fill-[#9b9392]"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
            />
          </svg>
          <div
            class="tooltip absolute flex gap-4 group-hover:visible"
          >
            <div
              class="flex flex-col"
            >
              <h2
                class="m-0 border-b py-2 text-sm font-medium text-black"
              >
                HALF SLEEVE T SHIRT
              </h2>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/half-sleeve-dawah-tshirts-for-men"
              >
                Half Sleeve Dawah T-Shirt
              </a>
            </div>
          </div>
        </div>
        <div
          class="group relative flex items-center"
        >
          <a
            class="text-sm font-medium text-[#f0f0f0] hover:text-[#f0f0f0]"
            href="/"
          >
            SUNNAH ESSENTIAL
          </a>
          <svg
            class="my-0 mx-1 w-[.6rem] fill-[#9b9392]"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
            />
          </svg>
          <div
            class="tooltip absolute flex gap-4 group-hover:visible"
          >
            <div
              class="flex flex-col"
            >
              <h2
                class="m-0 border-b py-2 text-sm font-medium text-black"
              >
                SUNNAH ESSENTIAL
              </h2>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/perfume"
              >
                Perfume
              </a>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/combo-perfume-box"
              >
                Combo Perfume Box
              </a>
              <a
                class="m-0 border-b py-2 text-base text-[#555555]"
                href="/prayermat"
              >
                Prayermat
              </a>
            </div>
          </div>
        </div>
        <div
          class="flex items-center"
          id="fabric-mask"
        >
          <a
            class="text-sm font-medium text-[#f0f0f0]"
            href="/fabric-mask"
          >
            FABRIC MASK
          </a>
        </div>
      </div>
    </div>
    <div
      class="mx-4 md:mx-12 lg:mx-16"
    >
      <div
        class="main mb-12 mt-9 lg:flex"
      >
        <div
          class="w-full lg:w-1/2"
        >
          <div
            class="relative w-full overflow-hidden lg:flex-1"
          >
            <img
              alt="Half Sleeve Cut and Sew Solid(pattern 15)"
              src="https://res.cloudinary.com/du3oueesv/image/upload/v1661588492/shopping%20cart/half%20sleeve%20cut/mthcssp15-solid-920x920_him69g.jpg"
            />
            <div
              style="display: none; position: absolute; pointer-events: none; height: 0px; width: 0px; top: 0px; left: 0px; opacity: 1; background-image: url(https://res.cloudinary.com/du3oueesv/image/upload/v1661588492/shopping%20cart/half%20sleeve%20cut/mthcssp15-solid-920x920_him69g.jpg); background-repeat: no-repeat; background-size: 0px 0px; background-position-x: 0px; background-position-y: 0px;"
            />
          </div>
        </div>
        <div
          class="mainInfo lg:ml-8"
        >
          <div
            class="breadcrumbs mt-2 text-sm lg:mt-0 lg:text-base"
          >
            <a
              href="/"
            >
              HOME / 
            </a>
            <a
              href="/half-sleeve-cut-and-sew-solid"
            >
              HALF SLEEVE CUT T-SHIRTS
            </a>
          </div>
          <h1
            class="mb-6 mt-2 text-xl font-bold text-[#555555] lg:text-3xl"
          >
            Half Sleeve Cut and Sew Solid(pattern 15)
          </h1>
          <h2
            class="mb-4 text-2xl font-bold text-slate-800"
          >
            ৳450.00
          </h2>
          <p
            class="mb-6"
          >
            <strong>
              Detailed Specification
            </strong>
          </p>
          <ul
            class="details list-disc"
          >
            <li
              class="mx-4 my-3"
            >
              Fabric: 100% Cotton
            </li>
            <li
              class="mx-4 my-3"
            >
              GSM: 160-165
            </li>
          </ul>
          <div
            class="fabricButtons"
          >
            <span
              class="product-option text-sm font-bold text-slate-800"
            >
              FABRIC
            </span>
            <br />
            <button
              class=" mb-1 mr-2 w-fit rounded-sm border-2 border-black bg-white p-1 text-base text-black"
            >
              COMBED COTTON
            </button>
          </div>
          <div
            class="sizeButtons"
          >
            <span
              class="product-option text-sm font-bold text-slate-800"
            >
              SIZE
            </span>
            <br />
            <button
              class=" mb-8 mr-2 w-fit rounded-sm border-2 border-black bg-white p-1 text-base text-black"
            >
              M
            </button>
            <button
              class=" mb-8 mr-2 w-fit rounded-sm border-2 border-black bg-white p-1 text-base text-black"
            >
              L
            </button>
            <button
              class=" mb-8 mr-2 w-fit rounded-sm border-2 border-black bg-white p-1 text-base text-black"
            >
              XL
            </button>
            <button
              class=" mb-8 mr-2 w-fit rounded-sm border-2 border-black bg-white p-1 text-base text-black"
            >
              XXL
            </button>
          </div>
          <div
            class="addToCart flex gap-4"
          >
            <div
              class="quantity"
            >
              <button
                class="h-10 w-8 bg-gray-200 hover:bg-slate-300"
              >
                -
              </button>
              <input
                class="h-10 w-8 border border-y-gray-200 text-center"
                type="number"
                value="1"
              />
              <button
                class="h-10 w-8 bg-gray-200 hover:bg-slate-300"
              >
                +
              </button>
            </div>
            <button
              class="w-fit bg-black px-6 py-2 font-bold text-white "
            >
              Add to Cart
            </button>
          </div>
        </div>
      </div>
      <div
        class="additional_info my-4"
      >
        <div
          class="additional_info_tabs md:border-top-gray-400 md:flex md:gap-2 md:border-t"
        >
          <button
            class="description active-tab peer border-t py-2 text-start font-bold hover:border-t-2 hover:border-t-black peer-hover:border-t peer-hover:border-t-gray-400"
          >
            Description
          </button>
          <br />
          <button
            class="additional-info peer border-t py-2 text-start font-bold hover:border-t-2 hover:border-t-black "
          >
            Additional Information
          </button>
        </div>
        <div
          class="additional_info_content my-4"
        >
          <p>
            <strong>
              Size Measurement (in inch/centimeter):
            </strong>
          </p>
          <ul>
            <li
              class="my-4"
            >
              M(Medium): Chest 39.37″ inch/ 100 cm, Height 27.55″ inch/ 70 cm
            </li>
            <li
              class="my-4"
            >
              L(large) : Chest 40.94″inch/ 104 cm, Height 28.34″inch/ 72 cm
            </li>
            <li
              class="my-4"
            >
              XL(Extra large): Chest 42.51″ inch/ 108 cm, Height 29.13″ inch/ 74 cm
            </li>
            <li
              class="my-4"
            >
              XXL(Extra large) : Chest 44.09″ inch/ 112 cm, Height 29.92″ inch/ 76 cm
            </li>
          </ul>
          <p
            class="my-4"
          >
            ডিভাইসের ভিন্নতা অনুযায়ী কালার ডেভিয়েশন এবং ফটোশপ এডিটিং কালার ডেভিয়েশনের কারনে বাস্তব ছবি ও সফটকপির মাঝে অনেক সময় পার্থক্য থেকে যায়
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
